version: '3.9'

services:
  db:
    image: postgres:13
    environment:
      POSTGRES_USER: chattyuser
      POSTGRES_PASSWORD: chattypassword
      POSTGRES_DB: chattydb
    volumes:
      - postgres_data:/var/lib/postgresql/data
    networks:
      - chatty_network

  redis:
    image: redis:alpine
    volumes:
      - redis_data:/data
    networks:
      - chatty_network

  django:
    build: .
    command: >
      sh -c "
      python manage.py makemigrations &&
      python manage.py migrate &&
      python manage.py collectstatic --no-input &&
      daphne -b 0.0.0.0 -p 8000 core.asgi:application
      "
    volumes:
      - .:/app
    environment:
      - DEBUG=False
      - HOST_IP=192.168.1.104
      - DJANGO_SECRET_KEY=django-insecure-t2p2d0psa55mhz8j&v-6_y@%i(vageo$r0^*g=(3dr^yop1o)*
      - DATABASE_URL=postgres://chattyuser:chattypassword@db:5432/chattydb
      - REDIS_URL=redis://redis:6379/0
    depends_on:
      - db
      - redis
    expose:
      - "8000"
    networks:
      - chatty_network

  nginx:
    image: nginx:alpine
    volumes:
      - ./nginx.conf:/etc/nginx/nginx.conf
      - ./static:/app/static
      - ./media:/app/media
    ports:
      - "80:80"
    depends_on:
      - django
    networks:
      - chatty_network

volumes:
  postgres_data:
  redis_data:

networks:
  chatty_network:
    driver: bridge
